gettsim/pre_processing/piecewise_functions.py

```diff
@@ -50,4 +61,17 @@ def piecewise_polynomial(
     out = intercept_interval
     for pol in range(1, rates.shape[0] + 1):
         out += rates[pol - 1, index_interval] * (increment_to_calc ** pol)
+
     return out
+
+
+piecewise_polynomial = np.vectorize(
```

Comment:
Would

```python
def piecewise_polynomial(...):
    return np.vectorize(...)
```
or something similar work? This way, we could have a proper docstring and would not run into problems with Sphinx.

Reply:
Sure, I can define it as a closure as you do. Or I can even write a full decorator which can receive excluded as an argument if we plan to have multiple similar functions like `piecewise_polynomial` in the future and decouple the vectorization from this function..

Reply:
Might be good to prepare for the latter usecase. 

I just took issue with the docstring. We should make sure that we only have one sensible one, which 
* stays clear of implementation details 
* and gets caught by Sphinx

Reply:
Changed it to the decorator. What do you mean by "gets caught by Sphinx"? Should I include it in the docs? Currently, there is no API section.

Reply:
Oh, we definitely want that eventually! So we should prepare for it, not necessary to implement it now.

Reply:
(sorry I had missed this one)